source/framework/vx_producer.c

```diff
@@ -0,0 +1,1336 @@
+    {
+        if (E_IPPC_OK == ippc_sender_receiver_ready(&producer->m_sender_ctx.m_sender, i) && 
+            (producer->consumers_list[i].state == PROD_STATE_CLI_NOT_CONNECTED))
+        {
+            producer->consumers_list[i].state       = PROD_STATE_CLI_CONNECTED;
+            producer->consumers_list[i].consumer_id = i;
+            producer->nb_consumers++;
+        }
+    }    
+}
+
+static void fill_reference_info(vx_producer producer, SProducerContent* buffid_message)
+{
+    buffid_message->num_refs = producer->numBufferRefsExport;
+    uint32_t i = 0;
+    uint32_t j = 0;
+    for (i = 0; i < producer->numBufferRefsExport; i++)
+    {
+        vx_enum                  ref_type;
+        vx_uint32                num_items          = 0;
+        tivx_utils_ref_ipc_msg_t ipc_message_parent = {0};
+        tivx_utils_ref_ipc_msg_t ipc_message_item[OVXGW_MAX_NUM_REFS];
+
+        vx_status framework_status =
+            vxQueryReference(producer->refs[i].ovx_ref, VX_REFERENCE_TYPE, (void*)&ref_type, (vx_size)sizeof(ref_type));
+        if (framework_status != VX_SUCCESS)
+        {
+            VX_PRINT(VX_ZONE_ERROR, "PRODUCER: vxQueryReference() failed for object [%d]\n", i);
+            break;
+        }
+        else if (ref_type == VX_TYPE_OBJECT_ARRAY)
+        {
+            framework_status = rbvx_utils_export_ref_for_ipc_xfer_objarray(
+                producer->refs[i].ovx_ref,
+                &num_items,
+                &ipc_message_parent,
+                (tivx_utils_ref_ipc_msg_t*)&ipc_message_item[0]);
+
+            buffid_message->num_items = num_items;
+
+            // send object array items data, if present
+            for (j = 0; j < num_items; j++)
+            {
+                memcpy(&buffid_message->ipc_msg[i][j], (void*)&ipc_message_item[j], sizeof(tivx_utils_ref_ipc_msg_t));
+
+                VX_PRINT(
+                    VX_ZONE_INFO,
+                    "PRODUCER %s: sending objarray element %d with fd count %d\n",
+                    producer->name,
+                    j,
+                    buffid_message->ipc_msg[i][j].numFd);
+            }
+        }
+        else
+        {
+            framework_status = tivx_utils_export_ref_for_ipc_xfer(producer->refs[i].ovx_ref, &ipc_message_parent);
+        }
+
+        if (framework_status != VX_SUCCESS)
+        {
+            VX_PRINT(VX_ZONE_ERROR, "PRODUCER: export_ref_for_ipc_xfer() failed for buffer [%d]\n", i);
+            break;
+        }
+        else
+        {
+            VX_PRINT(
+                VX_ZONE_INFO,
+                "PRODUCER %s: export of buffer successfull: %d of total: %d\n",
+                producer->name,
+                i + 1,
+                producer->numBufferRefsExport);
+        }
+
+        // send reference data, for object array this is final metadata
+        memcpy(&buffid_message->ipc_msg[i][j], &ipc_message_parent, sizeof(tivx_utils_ref_ipc_msg_t));
```

Comment:
same here, do not memcopy if not needed. This is a waste of runtime and CPU load

Reply:
Done